components/automate-cli/pkg/verifyserver/services/systemresourceservice/systemresourcesservice.go

```diff
@@ -0,0 +1,375 @@
+	FREE_SPACE_CHECK        = "%s free space check"
+
+	SUCCESS_MSG        = "%s should have free space >=%vGB"
+	ERROR_MSG          = "%s free space is %0.2fGB"
+	SUCCESS_MSG_IN_PER = " or %v%% of total size of /hab"
+
+	INVALID_NODE_TYPE_ERR = "given query node_type with value=%s is not supported"
+)
+
+func (srs *SystemResourcesService) GetSystemResourcesForDeployment(nodeType, deploymentState string) (*models.ApiResult, error) {
+	srsResponse := &models.ApiResult{
+		Passed: true,
+		Checks: []models.Checks{},
+	}
+
+	cpuCountCheck := srs.GetCpuCountCheck()
+
+	if !cpuCountCheck.Passed {
+		srsResponse.Passed = false
+	}
+
+	srsResponse.Checks = append(srsResponse.Checks, *cpuCountCheck)
+
+	cpuSpeedCheck := srs.GetCpuSpeedCheck(CPU_INFO_FILE)
+
+	if !cpuSpeedCheck.Passed {
+		srsResponse.Passed = false
+	}
+
+	srsResponse.Checks = append(srsResponse.Checks, *cpuSpeedCheck)
+
+	memorySizeCheck := srs.GetMemorySizeCheck()
+
+	if !memorySizeCheck.Passed {
+		srsResponse.Passed = false
+	}
+
+	srsResponse.Checks = append(srsResponse.Checks, *memorySizeCheck)
+
+	switch deploymentState {
+	case constants.PRE_DEPLOY:
+		habFreeSpaceCheckPreDepl, err := srs.GetHabFreeSpaceCheckPreDeployment(nodeType)
+		if err != nil {
+			return nil, err
+		}
+		if !habFreeSpaceCheckPreDepl.Passed {
+			srsResponse.Passed = false
+		}
+		srsResponse.Checks = append(srsResponse.Checks, *habFreeSpaceCheckPreDepl)
+
+	case constants.POST_DEPLOY:
+		habFreeSpaceCheckPostDepl, err := srs.GetHabFreeSpaceCheckPostDeployment(nodeType)
+		if err != nil {
+			return nil, err
+		}
+		if !habFreeSpaceCheckPostDepl.Passed {
+			srsResponse.Passed = false
+		}
+		srsResponse.Checks = append(srsResponse.Checks, *habFreeSpaceCheckPostDepl)
+	default:
+		srs.logger.Errorf("Wrong query deployment_state=%s", deploymentState)
+		return nil, fmt.Errorf("given query deployment_state with value=%s is not supported", deploymentState)
+	}
+
+	tmpFreeSpaceCheck := srs.GetFreeDiskSpaceCheckOfDir("/tmp", constants.TMP_FREE_DISK_IN_PER, constants.TMP_FREE_DISK_IN_GB, "Temp")
+
+	if !tmpFreeSpaceCheck.Passed {
+		srsResponse.Passed = false
+	}
+
+	srsResponse.Checks = append(srsResponse.Checks, *tmpFreeSpaceCheck)
+
+	rootFreeSpaceCheck := srs.GetFreeDiskSpaceCheckOfDir("/", constants.ROOT_FREE_DISK_IN_PER, constants.ROOT_FREE_DISK_IN_GB, "/(root volume)")
+
+	if !rootFreeSpaceCheck.Passed {
+		srsResponse.Passed = false
+	}
+	srsResponse.Checks = append(srsResponse.Checks, *rootFreeSpaceCheck)
+	return srsResponse, nil
+}
+
+func (srs *SystemResourcesService) GetCpuCountCheck() *models.Checks {
+	srs.logger.Debug("CPU count check is running")
+	cpuCount := srs.GetOsAndFileSystemInfo.GetNumberOfCPU()
+	srs.logger.Debug("CPU count is : ", cpuCount)
+
+	var checkResp *models.Checks
+	if cpuCount >= constants.MIN_CPU_COUNT {
+		successMsg := fmt.Sprintf("CPU count is >=%v", constants.MIN_CPU_COUNT)
+		checkResp = srs.GetChecksModel(true, CPU_COUNT_CHECK_TITLE, successMsg, "", "")
```

Comment:
return on line 123 and avoid else. I think you even do not need a variable just return the function call

Reply:
Done